dask_remote/client_base.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional
+from urllib import parse
+
+from requests import Session, auth
+
+
+# Default connection timeout at 10''
+DEFAULT_TIMEOUT = 10.0
+
+# Default JSON response back
+DEFAULT_HEADERS = {"Accept": "application/json"}
+
+
+class ClientError(Exception):
+    ...
+
+
+class JSONClient:
+    """Base class for JSON clients."""
+
+    def __init__(
+        self, base_url: str, default_timeout: float = None, default_headers: dict = None
+    ) -> None:
+        self._url = base_url
+        self._timeout = default_timeout or DEFAULT_TIMEOUT
+        self._headers = default_headers or DEFAULT_HEADERS
+        self._connect()
+
+    # Public HTTP methods:
+
+    def get(self, path: str, params: Optional[dict] = None) -> dict:
+        return self._request(method="GET", path=path, params=params)
+
+    def post(self, path: str, params: Optional[dict] = None, data: Optional[dict] = None) -> dict:
+        return self._request(method="POST", path=path, params=params, data=data)
+
+    # Private methods:
+
+    def _connect(self) -> None:
+        """Connect to the resource."""
+        self._conn = Session()
```

Comment:
see above, `self.conn` could return a Session, with `self._conn = None` in _init_